src/Library/demos/Preferences Window/main.blp

```diff
@@ -0,0 +1,160 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.PreferencesWindow pref_window {
+  default-width: 800;
+  default-height: 600;
+  title: "My App Preferences";
+
+  Adw.PreferencesPage appearance_page {
+    title:"Appearance";
+    icon-name:"brush-monitor-symbolic";
+
+    Adw.PreferencesGroup{
+      title:"Color Settings";
+      description:"Change the color-scheme of the application.";
+
+
+      Adw.ActionRow{
+        title: "Use Dark Mode";
+        activatable-widget: dm_switch;
+
+        [suffix]
+        Switch dm_switch{
+          halign: center;
+          valign: center;
+        }
+      }
+    }
+
+    Adw.PreferencesGroup {
+      title: "Text Settings";
+      description: "Customize the appearance of text in the application.";
+      //
+
+      Adw.ActionRow {
+        title: "Font Size";
+
+        [suffix]
+        Gtk.SpinButton{
+          halign: center;
+          valign: center;
+          adjustment:
+          Gtk.Adjustment size_adj {
+            lower: 5;
+            upper: 999;
+            step-increment: 1;
+            value: 11;
+          };
+        }
+      }
+
+      Adw.ActionRow {
+        title: "Font Color";
+
+        [suffix]
+        Gtk.ColorDialogButton{
+          halign: center;
+          valign: center;
+          dialog: ColorDialog {};
+        }
+      }
+
+      LinkButton {
+        label: "API Reference";
+        uri: "https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1.3/class.PreferencesWindow.html";
+      }
```

Comment:
please add a margin above, similar to other demos

I don't think we'll have separate entries for related widgets

We can add links for

* PreferencesWindow API Reference
* PreferencesPage API Reference
* PreferencesGroup API Reference
* PreferencesRow API Reference